spec/rspec/rails/matchers/has_spec.rb

```diff
@@ -1,27 +1,31 @@
 require 'spec_helper'
 
 class CollectionOwner < ActiveRecord::Base
-  connection.execute <<-SQL
-    CREATE TABLE collection_owners (
-      id integer PRIMARY KEY AUTOINCREMENT
-    )
-SQL
   has_many :associated_items do
     def has_some_quality?; true end
   end
 end
 
 class AssociatedItem < ActiveRecord::Base
-  connection.execute <<-SQL
-    CREATE TABLE associated_items (
-      id integer PRIMARY KEY AUTOINCREMENT,
-      collection_owner_id integer
-    )
-SQL
   belongs_to :collection_owner
 end
 
 describe "should have_xxx" do
+  before(:all) do
+    CollectionOwner.connection.execute <<-SQL
+      CREATE TABLE collection_owners (
+        id integer PRIMARY KEY AUTOINCREMENT
+      )
+    SQL
+
+    AssociatedItem.connection.execute <<-SQL
+      CREATE TABLE associated_items (
+        id integer PRIMARY KEY AUTOINCREMENT,
+        collection_owner_id integer
+      )
+    SQL
+  end
```

Comment:
Why have you moved these?

Reply:
Because of the randomly running tests, the memory sqlite connection/spec is going to be removed. That's why the table is missing sometimes in the middle of the run. The SQL migration was inlined to the example group due to this fact.